istioctl/cmd/kubeutils.go

```diff
@@ -0,0 +1,169 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"istio.io/pkg/env"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubectl/pkg/scheme"
+)
+
+// k8sClientConfig first tries to get the config from inside a kubernetes environment
+// If unsuccessful, it tries to get the config from the default kubeconfig file
+func k8sClientConfig() (*rest.Config, error) {
+	config, err := rest.InClusterConfig()
+	var ocErr error
+	if err != nil {
+		config, ocErr = clientcmd.BuildConfigFromFlags("", kubeconfigPath())
+		if ocErr != nil {
+			return nil, fmt.Errorf("in-cluster error: %v, out-of-cluster error: %v", err, ocErr)
+		}
+	}
+	return config, nil
+}
+
+// Returns a kubernetes clientset with the default config
+func k8sClientset() (*kubernetes.Clientset, error) {
+	config, err := k8sClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return clientset, nil
+}
+
+// Returns a kubernetes service in a namespace
+// TODO: remove once merged
+// nolint: deadcode
+func k8sService(name, namespace string) (*v1.Service, error) {
+	clientset, err := k8sClientset()
+	if err != nil {
+		return nil, err
+	}
+	service, err := clientset.CoreV1().Services(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+// Returns a kubernetes pod in a namespace
+func k8sPod(name, namespace string) (*v1.Pod, error) {
+	clientset, err := k8sClientset()
+	if err != nil {
+		return nil, err
+	}
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, name) {
+			return &pod, nil
+		}
+	}
+	return nil, fmt.Errorf("pod with name %s not found in namespace %s", name, namespace)
+}
+
+// Returns a kubernetes config map in a namespace
+// TODO: remove once merged
+// nolint: deadcode
+func k8sConfigMap(name, namespace string) (*v1.ConfigMap, error) {
+	clientset, err := k8sClientset()
+	if err != nil {
+		return nil, err
+	}
+	cm, err := clientset.CoreV1().ConfigMaps(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return cm, nil
+}
+
+// Execs a command inside a kubernetes pod and returns stdout and stderr
+// nolint: unparam
+func k8sPodsExec(name, namespace, command string, stdin io.Reader) (sout, serr string, err error) {
```

Comment:
istioctl already has code to exec into a pod afaik, strongly prefer to re-use this